src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -520,11 +501,17 @@ public BitbucketApi buildBitbucketClient(String repoOwner, String repository) {
     @Override
     public void afterSave() {
```

Comment:
This override might not be necessary at all. It used to ensure that both `repositoryType` and `cloneLinks` were set. However, you deleted `repositoryType`, and `cloneLinks` is only used in `build(SCMHead head, SCMRevision revision)`, which itself gets the clone links from Bitbucket if not already set. Before your changes, the override might have been necessary because some methods that read `repositoryType` did not expect null there.

I'm not sure whether deleting the override entirely breaks binary compatibility in Java. If it does, then keep the override and just call super from there.

Or is this necessary for scenarios where a previously initialised `cloneLinks` list is now out of date — because the SSH port number of the Bitbucket server has been changed, or because `build(SCMHead head, SCMRevision revision)` was not able to query the links and made incorrect assumptions? Perhaps the Jenkins administrator used to be able to recover from that situation by saving the project configuration again in Jenkins. That seems unlikely to have worked, though, because `repositoryType` would usually have been non-null and prevented `afterSave` from making the request.

Reply:
As far i understood, this method will be called after job saving. Previous code fetched links, if new job was created, The logic behind was: There is new job available, repository type was changed --> fetch links, if possible. If already existing job was modified, then no action was performed. Now it's the same logic, excepting the guard is `cloneLinks` and not `repositoryType`. The question now is: Shall the links be fetched again, if job was changed. From my point of view, yes. I would implement it as:
```
try {
    BitbucketRepository r = buildBitbucketClient().getRepository();
    Map<String, List<BitbucketHref>> links = r.getLinks();
    if (links != null && links.containsKey("clone")) {
        cloneLinks = links.get("clone");
    }
} catch (InterruptedException | IOException e) {
    LOGGER.log(Level.SEVERE,
    "Could not determine clone links of " + getRepoOwner() + "/" + getRepository() +
    " on " + getServerUrl() + " for " + getOwner() + " falling back to generated links", e);
}
```
i.e. without `if`. FDYT?